index.ts

```diff
@@ -44,21 +44,30 @@ export interface IJolocomSDKInitOptions {
   dontAutoRegister?: boolean
 }
 
-// @ts-ignore
-// const injectPassFn = (passFn: () => Promise<string>) => <A, T>(
-//   delayedFn: (arg1: A, pass: string, ...rest) => Promise<T>,
-// ) => async (a1: A, ...rest) => await delayedFn(a1, await passFn())
-
-export class JolocomSDK {
+export class JolocomSDK extends BackendMiddleware {
+  // FIXME merge the backendMiddleware code in here instead of extending??
+  //       or perhaps the BackendMiddleware becomes the more "pure" layer,
+  //       and the "sdk" instance is platform specific
   public bemw: BackendMiddleware
 
+  /* TODO: use the Keeper pattern
+   **      so we can do sdk.identities.create()
+                        sdk.interactions.create()
+                        sdk.interactions.find()
+                        etc
+  identities: IdentityKeeper
+  interactions: InteractionKeeper
+
+  /**/
+
   constructor(conf: IJolocomSDKConfig) {
-    this.bemw = new BackendMiddleware({
+    super({
       ...defaultConfig,
       storage: conf.storage,
-      passwordStore: conf.passwordStore,
-      sdk: this
+      passwordStore: conf.passwordStore
     })
+    // FIXME remove
+    this.bemw = this
```

Comment:
While this does preserve the API, we're still in a breakable API phase so I dont think that's enough of a concern to support a pattern like this :laughing:. Also, while this might support the SDK useage in the wallet, I doubt there's a need to use the `bemw` member basically ever, especially as the identity wallet is exposed via .idw(). I'd rather this be a breaking change where the backendmiddleware is removed/collapsed entirely

Reply:
it's gonna need some quick find and replace in the SW2.0 to readjust to this, that's really the only consumer right now
cc @clauxx @SBub

Reply:
Will remove this and actually update the SW!